src/pages/Dashboard/Components/TimeEntryForm/TimeEntryForm.js

```diff
@@ -138,6 +139,8 @@ const mapStateToProps = (state) => {
 const mapDispatchToProps = (dispatch, ownProps) => {
   return {
     createTimeEntry: (timeEntry) => {
+      timeEntry.start = timeEntry.start.format(Configuration.dateFormat);
+      timeEntry.end = timeEntry.end.format(Configuration.dateFormat);
```

Comment:
Los `.format` deberian hacerlos quien invocan la acción. `mapDispatchToProps` deberia retornar la acción en si misma, y no un wrapper.

Esto deberia quedar:
```js
const mapDispatchToProps = {
  createTimeEntry
};
```

y en `stopTimer: 
```js
const { start, projectId } = this.state;

this.props.createTimeEntry({
  start: start.format(),
  end: moment().format()
  projectId,     
  taskName: this.taskNameInput.value	
});
```

En esto hay que notar:

- La respuesta por defecto de `moment().format()` es valida para nosotros. No hay necesidad de pasar el formato
- Para los elementos de un form hay que usar `ref` (ver el link que te copie varios threads atras). Ahi es donde se crea `this.taskNameInput`